### PR TITLE
Feature/shrinking radius

### PR DESCRIPTION
**Pull request by Tesa27**
Have a radius around in which the coins are generated the shrinks. And fixed a small problem of coin display for non-hosts

### CODE REVIEW

File: app/src/androidTest/java/sdp/moneyrun/map/MapInstrumentedTest.java [hunk 1]
```diff
@@ -1531,7 +1531,15 @@ public void onDidFinishRenderingMap(boolean fully) {
 
         }
     }
-
+    @Test
+    public void checkIfShrinkCircleDrawsSmallerCircleWithCorrectShrinkPace(){
```
Comment (gauthierboeshertz) [inline app/src/androidTest/java/sdp/moneyrun/map/MapInstrumentedTest.java:1535]: This doesnt really test anything

**Review by gauthierboeshertz — CHANGES_REQUESTED** [review]

File: app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java [hunk 2]
```diff
@@ -499,4 +509,76 @@ public void checkObjectives(Location location) {
...
+     * @param shrinkPace bz how much the radius shrinks
+     * @param epsilon if under this threshold there is no circle anymore
+     */
+
+    public void shrinkCircle(long paceInMillis, double initialRadius,LatLng center,int numberOfSides,MapboxMap map,double shrinkPace,double epsilon){
+        drawCircle(center,initialRadius,numberOfSides,map);
+        if(initialRadius < epsilon)
+            return;
+        long initTime = currentTimeMillis();
+        while(currentTimeMillis() - initTime < paceInMillis){}
```
Comment (gauthierboeshertz) [inline app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java:528]: Blocking the activity thread for some time looks like a bad practice

**Review by gauthierboeshertz — CHANGES_REQUESTED** [review]

**Review by gauthierboeshertz — APPROVED** [review]

File: app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java [hunk 3]
```diff
@@ -229,10 +229,10 @@ private void createJoinButton(Button button, int buttonId, GameRepresentation ga
 
                     double distance = gameRepresentation.getStartLocation().distanceTo(locationRep);
 
-                    if (distance > MAX_DISTANCE_TO_JOIN_GAME) {
-                        button.setEnabled(false);
-                        button.setText(activity.getString(R.string.join_game_too_far_message));
-                    }
+//                    if (distance > MAX_DISTANCE_TO_JOIN_GAME) {
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java:232]: Comments should be removed

File: app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java [hunk 4]
```diff
@@ -497,6 +530,20 @@ public void checkObjectives(Location location) {
         }
 
     }
+    public void initCircle(){
+        circleManager.deleteAll();
+        CircleOptions circleOptions = new CircleOptions();
+        circleOptions = circleOptions.withCircleRadius(circleRadius);
+        circleOptions = circleOptions.withCircleOpacity(0.5f);
+        circleOptions = circleOptions.withCircleColor(""+Color.blue(8));
+        //circleOptions.withCircleColor(ColorUtils.colorToRgbaString(getResources().getColor(R.color.colorPrimary)));
```
Comment (Alvaro2112) [inline app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java:539]: Same here

**Review by Alvaro2112 — CHANGES_REQUESTED** [review]
I think you should add Javadoc to the functions you coded, also, have you use the reformatter and the code analysis tool to fix any warnings?

File: app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java (hunk 4 above)
Comment (Tesa27) [reply app/src/main/java/sdp/moneyrun/ui/map/MapActivity.java:557]: done

File: app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java (hunk 3 above)
Comment (Tesa27) [reply app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java]: done

**Review by Alvaro2112 — APPROVED** [review]

**Pull request merged**